liquidcore/twofactor/views.py

```diff
@@ -19,7 +21,24 @@ def get_png(user, device):
 
 @transaction.atomic
 def invitation(request, code):
-    invitation = invitations.get_or_404(code)
+    invitation = invitations.get(code)
+
+    if not invitation:
+        return render(request, 'totp-invitation-nonexistent.html')
+
+    if invitation.state == 'expired':
+        return render(request, 'totp-invitation-expired.html',
+                      {'timeout': settings.LIQUID_2FA_INVITATION_VALID})
+    if invitation.state == 'used':
+        return render(request, 'totp-invitation-used.html')
+    if invitation.state == 'valid' and request.user.is_authenticated:
+        return render(request, 'totp-invitation-loggedin.html',
```

Comment:
and 403 again

all these should be non-200 to signal that these are abnormal conditions, mostly for ourselves when testing

user should still see all pages in browser with no visible change

you know what they say - ![](https://i.kym-cdn.com/photos/images/facebook/000/918/810/a22.jpg)